lib/api/bucketPutACL.js

```diff
@@ -281,12 +285,15 @@ export default function bucketPutACL(authInfo, request, log, callback) {
         if (err) {
             log.trace('error processing request', { error: err,
                 method: 'bucketPutACL' });
-            return callback(err);
+            if (err === errors.NoSuchBucket) {
+                return callback(err);
+            }
```

Comment:
Since I pass the bucket through the waterfall with https://github.com/scality/S3/pull/541/commits/eed031365b423d65ce9657f9b14797cd28beaf0d, I don't need the check for NoSuchBucket err any more (just need to not pass the bucket when the error is NoSuchBucket and there won't be any corsHeaders).